receiver/mysqlreceiver/testdata/scraper/expected.json

```diff
@@ -4,7 +4,7 @@
          "instrumentationLibraryMetrics": [
             {
                "instrumentationLibrary": {
-                  "name": "otel/mysql"
+                  "name": "otelcol/mysqlreceiver"
```

Comment:
Shouldn't this be a part of another PR?

Reply:
If i change the Emit interface I end up having to change the receiver that weren't already refactored to use `NewMetricData`. I can migrate the receivers (mysql, zookeeper) to NewMetricData as a middle step to this change.

Reply:
> I can migrate the receivers (mysql, zookeeper) to NewMetricData as a middle step to this change.

I though that was the plan. But it's up to you

Reply:
I vote for splitting it just to remove "breaking changes" from this PR to keep it cleaner.

Reply:
I split them into #8387 and #8389 👍

Reply:
Rebased.